internal/buffer/settings.go

```diff
@@ -55,7 +55,26 @@ func (b *Buffer) SetOptionNative(option string, nativeValue interface{}) error {
 				buf.HighlightSearch = nativeValue.(bool)
 			}
 		}
-	}
+	} else {
+		for _, pl := range config.Plugins {
+			if option == pl.Name {
+				if nativeValue.(bool) {
+					if !pl.Loaded {
+						pl.Load()
+					}
+					_, err := pl.Call("init")
```

Comment:
init and deinit are a global thing, right? Calling them has effect on all buffers.

Reply:
On more thought, this concerns the above `pl.Load()` call as well, since it enables `onBufferOpen`, `onBufPaneOne` etc, which are global.

This seems be a problem with the very concept of enabling/disabling plugins per buffer, not just with its implementation in this PR. If a plugin is disabled globally (and thus not loaded yet), and we enable it locally for a given buffer, we need to load this plugin, to make it work for this buffer. But by loading it we actually enable it globally for other buffers as well (although only partially, but that's even worse).

Reply:
> init and deinit are a global thing, right? Calling them has effect on all buffers.

> On more thought, this concerns the above pl.Load() call as well, [...]

Unfortunately, yes. :worried:
The focus was on preventing the usual callbacks.

Reply:
And the result seems buggy. The user may run `setlocal foo on` and expect it not to take any effect for other buffers, but it does take effect for them.

Maybe, to avoid that, we shouldn't load the plugin when enabling it locally, i.e. assume that it is already initialized globally, i.e. that it is enabled globally. Which would probably also mean that we should not allow enabling the plugin locally if it is not enabled globally (and should print an error to the user in this case).

I don't quite like this suggestion myself, but at least it would avoid unexpected unwanted side effects in the most common cases.

Looking at the `autoclose` plugin as an example, it so happens that it can be safely enabled/disabled without limitations, since it only includes "local" callbacks and no global state (it has global variables but they are actually constant). But we cannot assume that about an arbitrary plugin, and most plugins are not like that.